universal-application-tool-0.0.1/app/repository/ApplicantRepository.java

```diff
@@ -34,6 +39,32 @@ public ApplicantRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exe
         () -> ebeanServer.find(Applicant.class).setId(id).findOneOrEmpty(), executionContext);
   }
 
+  public CompletionStage<ImmutableList<ProgramDefinition>> programsForApplicant(long id) {
+    return supplyAsync(
+            () ->
+                ebeanServer
+                    .find(Program.class)
+                    .alias("p")
+                    .where()
+                    .or()
+                    .eq("lifecycle_stage", LifecycleStage.ACTIVE)
+                    .exists(
+                        ebeanServer
+                            .find(Application.class)
+                            .where()
+                            .eq("applicant.id", id)
+                            .raw("program.id = p.id")
+                            .query())
+                    .endOr()
+                    .findList(),
+            executionContext.current())
+        .thenApplyAsync(
+            (programs) ->
+                programs.stream()
+                    .map(program -> program.getProgramDefinition())
+                    .collect(ImmutableList.toImmutableList()));
```

Comment:
```suggestion
                    .collect(ImmutableList.toImmutableList()), httpExecutionContext.current());
```

Reply:
Don't they use the previous execution context if one isn't provided?  Let me check on that - that's been my belief so far, so I might have some cleaning up to do if not...

Reply:
Yeah, that's my read:

> Returns a new CompletionStage that, when this stage completes normally, is executed using this stage's default asynchronous execution facility, with this stage's result as the argument to the supplied function.

"this stage" in this context is "the previous stage" - this is the documentation for `thenApplyAsync`, so `this` is the previous stage.

Reply:
Oh, I see, you want to use a different context - the http context for mapping to a program definition.  Sure, I can plumb that through.

Reply:
Hmm, I'm actually going to go back on that - none of the repositories use the http context yet, this would be the first time.  Should I still, do you think?

Reply:
It's a pretty small amount of code that does no i/o so leaving it on the DB threadpool seems fine I guess.